connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/InboundConnectorBeanDefinitionProcessor.java

```diff
@@ -0,0 +1,129 @@
+import java.util.Arrays;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.ListableBeanFactory;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanDefinitionRegistryPostProcessor;
+
+public class InboundConnectorBeanDefinitionProcessor
+    implements BeanDefinitionRegistryPostProcessor {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(InboundConnectorBeanDefinitionProcessor.class);
+  private final List<InboundConnectorProperties> preparedProperties = new ArrayList<>();
+
+  @Override
+  public void postProcessBeanDefinitionRegistry(BeanDefinitionRegistry registry)
+      throws BeansException {
+    ListableBeanFactory listableBeanFactory = (ListableBeanFactory) registry;
+    String[] handlerBeans =
+        listableBeanFactory.getBeanNamesForType(InboundConnectorExecutable.class, true, false);
+    LOG.info("Found inbound connector beans: {}", (Object) handlerBeans);
+    for (String beanName : handlerBeans) {
+      BeanDefinition beanDefinition = registry.getBeanDefinition(beanName);
+      InboundConnector inboundConnectorAnnotation =
+          listableBeanFactory.findAnnotationOnBean(beanName, InboundConnector.class);
+      if (inboundConnectorAnnotation != null) {
+        InboundConnectorProperties properties =
+            getProperties(inboundConnectorAnnotation, beanDefinition, beanName);
+        preparedProperties.add(properties);
+      }
+    }
+  }
+
+  private InboundConnectorProperties getProperties(
+      InboundConnector inboundConnector, BeanDefinition beanDefinition, String beanName) {
+    var scope = beanDefinition.getScope();
+    if (!SCOPE_PROTOTYPE.equals(scope)) {
+      throw new IllegalStateException(
+          "Only \""
+              + SCOPE_PROTOTYPE
+              + "\" scope is supported for inbound connectors but found: \""
+              + scope
+              + "\" for bean: "
+              + beanDefinition);
+    } else {
+      InboundConnectorProperties properties =
+          getInboundConnectorProperties(inboundConnector, beanDefinition, beanName);
+      return properties;
+    }
+  }
+
+  private InboundConnectorProperties getInboundConnectorProperties(
+      InboundConnector inboundConnector, BeanDefinition beanDefinition, String beanName) {
+    var deduplicationProperties = Arrays.asList(inboundConnector.deduplicationProperties());
+    Class<? extends InboundConnectorExecutable> inboundConnectorClass;
+    try {
+      inboundConnectorClass =
+          (Class<? extends InboundConnectorExecutable>)
+              Class.forName(beanDefinition.getBeanClassName());
```

Comment:
Is this really necessary? You know what class these beans have because you asked the factory for beans of type `InboundConnectorExecutable`. You do not need to know the concrete class, it is an `InboundConnectorExecutable` and that is all you need to know. The property `clazz` of `InboundConnectorProperties` could then be removed as well. In addition, `getBeanClassName` [may not even return the actual class name](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/config/BeanDefinition.html#getBeanClassName()).

Reply:
I removed this as the instance creation with a customSupplier does not require the class to be present at all